kuma/users/signal_handlers.py

```diff
@@ -27,6 +71,16 @@ def on_user_signed_up(sender, request, user, **kwargs):
             )
 
 
+@receiver(user_logged_in, dispatch_uid='users.user_logged_in')
+def on_user_logged_in(sender, request, user, **kwargs):
+    sociallogin = kwargs.get('sociallogin')
+    if sociallogin and not getattr(request, 'signed_up', False):
```

Comment:
We need to check for the `social_auth_added` attribute here as well, so these two lines should change to something like:
```python
if getattr(request, 'signed_up', False) or getattr(request, 'social_auth_added', False):
    return
sociallogin = kwargs.get('sociallogin')
if sociallogin:
```